src/domain/flow-system/services/src/flow/flow_service_impl.rs

```diff
@@ -101,6 +102,44 @@ impl FlowServiceImpl {
         Ok(rounded_time)
     }
 
+    async fn load_flow(&self, flow_id: FlowID) -> Result<Flow, LoadError<FlowState>> {
+        DatabaseTransactionRunner::run_transactional_with(
+            &self.catalog,
+            |event_store: Arc<dyn FlowEventStore>| async move {
+                Flow::load(flow_id, event_store.as_ref()).await
+            },
+        )
+        .await
+    }
+
+    async fn save_flow(&self, flow: &mut Flow) -> Result<(), InternalError> {
+        DatabaseTransactionRunner::run_transactional_with(
+            &self.catalog,
+            |event_store: Arc<dyn FlowEventStore>| async move {
+                flow.save(event_store.as_ref()).await.int_err()?;
+
+                Ok(())
+            },
+        )
+        .await
+    }
```

Comment:
I think we should control transactions in `FlowService` at a bit higher level.
1) Definitely a transaction on startup, which has to load initial configurations and do the planning.
2) 1 transaction per 1 call of `run_current_timeslot`. Scheduling is not really an expensive operation, we are loading a few flows and inserting tasks into a queue.
3) 1 transaction per event in `AsyncEventHandler`s. These are coming independently from `TaskExecutor`. Make sure events are only sent when task executor's own transaction is not open.

Double check if we have any long-running computations here in between, but I don't think so, as `FlowService` mostly coordinates execution.

Queries like `list_all_flows_by_dataset` are already runnning behind a transaction in HTTP middleware.